metadata/src/main/java/org/apache/kafka/controller/BrokersToIsrs.java

```diff
@@ -191,7 +191,7 @@ void update(Uuid topicId, int partitionId, int[] prevIsr, int[] nextIsr,
     void removeTopicEntryForBroker(Uuid topicId, int brokerId) {
         Map<Uuid, int[]> topicMap = isrMembers.get(brokerId);
         if (topicMap != null) {
-            if (brokerId == NO_LEADER) {
+            if (brokerId == NO_LEADER && topicMap.containsKey(topicId)) {
                 offlinePartitionCount.set(offlinePartitionCount.get() - topicMap.get(topicId).length);
```

Comment:
Does this mean the `offlinePartitionCount` is no longer accurate? we don't update `offlinePartitionCount` when we delete a topic.

Reply:
We do update `offlinePartitionCount` when we delete a topic, right? Check `ReplicationControlManager#replay(RemoveTopicRecord record)`

The change here is to avoid a null pointer exception if `topicMap.get(topicId) == null`

Seems like a good fix

Reply:
That's right, @cmccabe. We still correctly update the offline partition count.